src/com/lftechnology/java/training/alina/jdbc/dao/user/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,220 @@
+        User user = new User();
+        LoginView.displayLoginHeader();
+        user = UserService.setLoginInfo(scanner, Constants.USER_LOGIN);
+        Boolean loginStatus = false;
+        Map<Integer, Object> params = new HashMap<Integer, Object>();
+        params.put(1, user.getUsername());
+        params.put(2, user.getPassword());
+        params.put(3, Constants.NOT_TERMINATED);
+        params.put(4, Constants.NOT_DELETED);
+        Database database = UserService.setDatabaseParams(params);
+        String sql =
+                "SELECT * FROM user u inner join employee e on (u.user_id = e.user_id) where u.username=? and u.password=? and u.is_terminated=? and e.is_deleted=?";
+        Connection connection = DbFacade.getDbConnection();
+        PreparedStatement preparedStatement = connection.prepareStatement(sql);
+        preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+        ResultSet result = preparedStatement.executeQuery();
+        if (result.next()) {
+            loginStatus = true;
+            checkValidUser(scanner, result.getInt("user_id"), connection);
+        } else {
+            loginStatus = false;
+        }
+        return loginStatus;
+    }
+
+    /**
+     * Checks if user is valid
+     * 
+     * @param scanner
+     *            {@link Scanner}
+     * @param userId
+     *            {@link Integer}
+     * @param connection
+     *            {@link Connection}
+     * @throws SQLException
+     * @author Alina Shakya <alinashakya@lftechnology.com>
+     */
+    private void checkValidUser(Scanner scanner, int userId, Connection connection) throws SQLException {
+        EmployeeDaoImpl employeeDao = new EmployeeDaoImpl();
+        UserService userService = new UserService();
+        List<Employee> employeeDetails = employeeDao.findByPk(userId);
+        for (Employee employee : employeeDetails) {
+            userService.getEmployeeRole(scanner, employee);
+        }
+    }
+
+    @Override
+    public List<User> findByPk(Integer userId) {
+        List<User> userList = new ArrayList<User>();
+        Map<Integer, Object> params = new HashMap<Integer, Object>();
+        params.put(1, userId);
+        Database database = UserService.setDatabaseParams(params);
+        String sql = "Select * from user where user_id=?) ";
+        try {
+            Connection connection = DbFacade.getDbConnection();
+            PreparedStatement preparedStatement = connection.prepareStatement(sql);
+            preparedStatement = DbFacade.setParameterizedObjects(database, preparedStatement);
+            ResultSet result = preparedStatement.executeQuery();
+            while (result.next()) {
+                UserService userService = new UserService();
+                userList.add(userService.map(result));
+            }
+            preparedStatement.close();
+        } catch (SQLException sqe) {
+            LOGGER.log(Level.WARNING, "SQLException : {0}", new Object[] { sqe });
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "SQLException : {0}" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)